src/Script.sol

```diff
@@ -12,13 +25,8 @@ import {StdStorage, stdStorageSafe} from "./StdStorage.sol";
 import {StdUtils} from "./StdUtils.sol";
 import {VmSafe} from "./Vm.sol";
 
-abstract contract ScriptBase is CommonBase {
-    // Create2 factory used by scripts when deploying with create2, https://github.com/Arachnid/deterministic-deployment-proxy.
-    address internal constant CREATE2_FACTORY = 0x4e59b44847b379578588920cA78FbF26c0B4956C;
-
-    VmSafe internal constant vmSafe = VmSafe(VM_ADDRESS);
-}
-
+// ✨ SCRIPT DX
```

Comment:
Nit, "Script Developer Experience" doesn't make much sense as a contract header comment 😅
```suggestion
// ✨ SCRIPT CONTRACT
```

Reply:
Okay, but these are not contract headers, they are there to form a template of what a `Script.sol` file should look like.

Reply:
Any other ideas?

Reply:
> Okay, but these are not contract headers, they are there to form a template of what a Script.sol file should look like.

Maybe I'm misunderstanding what you're trying to convey here. The intent of `Script.sol` is for people to inherit and use in their scripts. I don't think we have a good reason to encourage people to write their own `Script.sol` files and use them instead of this one.

It's great that these contracts modular and people _can_ do that, but these Script and Test contracts can be used directly and that's what forge-std is primarily intended to enable. This contract's docs should be explaining how to use this file in your scripts (and same for test). Docs on how to build your own `Script.sol` file modularly should live elsewhere (the README, the book, etc)

Reply:
For any sufficiently complex project, you have to customize `Test`.

 > these Script and Test contracts can be used directly and that's what forge-std is primarily intended to enable

Yes. That's why `Test.sol` and `Script.sol` are there - to provide the default testing/scripting setup. But, as I intentionally made the files looks like, there's nothing special in them, and anyone can make their own if they need to. If they don't, they are not expected to. I can communicate it better in the docs.

Reply:
I think the wording is already clear?

`You can customize it to suit the needs of your project by creating a similar file`
Or `If required, you can...`

Re docs: They can be improved, but I think they should stay in the file. Let me know if you disagree.

Reply:
> For any sufficiently complex project, you have to customize `Test`.

What do you mean by customizing `Test`? Even for complex projects I just inherit from `Test` then add any additional helpers I need, but I've never needed to modify the `Test` base contract itself

Reply:
What if you want to replace `DSTest` or `StdAssertions`?

Reply:
> You can customize it to suit the needs of your project by creating a similar file

I don't think we should suggest this IMO. I find that sentence confusing without more docs/context/rationale (i.e. "why would I want to do that? what am I missing?"), and I can't think of a good reason for doing that vs. just inheriting from `Script` or `Test` and adding methods you need.

forge-std is written modularly so people can do whatever they want, but forge-std shouldn't encourage breaking it apart unless we have a good reason to encourage that, and I don't think we do (which is good! because we want it to be dead-simple, drop-in usage—if you have to customize it, something is wrong with forge-std)

If there are good arguments for replacing DSTest or StdAssertions, those changes should be upstreamed to forge-std, but so far I don't know of any good arguments for that 🙂